uast/tonoder.go

```diff
@@ -86,6 +95,22 @@ func (c *BaseToNoder) toNode(obj interface{}) (*Node, error) {
 	}
 
 	n := NewNode()
+
+	// We need to have the internalkey before iterating others
+	internalKey, err := c.getInternalKeyFromObject(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	var promotedKeys map[string]bool
+	if (!c.PromoteAllPropertyLists) {
+		promotedKeys = c.PromotedPropertyLists[internalKey]
```

Comment:
Actually I tested  (and is nil on the first unittest) and:

- First access on the parent map when is nil: returns a "nil" map[string]bool
- Second access on this one when is nil (the line above): returns the default boolean value, false.

Check this:
https://play.golang.org/p/J4U7ipI7vn

But adding a nil check is not a bad idea anyway, I'll also add an aditional unittest because I just saw that I don't have a case for PromotedPropertyLists = nil and PromoteAllPropertyLists = true.